bin/cucc.in

```diff
@@ -0,0 +1,287 @@
+    # constexpr`` functions, and device code to invoke ``__host__
+    # constexpr`` functions.
+    # [NVCC v12.2 4.2.7.7]
+    #
+    # In HIP constexpr functions can be called from __device__ and
+    # __host__ code. Not sure if this option means to allow:
+    #   * __device__ -> __host__ constexpr calls and
+    #   * __host__ -> __device__ constexpr calls.
+    parser.add_argument("-expt-relaxed-constexpr", "--expt-relaxed-constexpr",
+                        action='store_true')
+
+    # "Allow __host__, __device__ annotations in lambda declarations."
+    # [NVCC v12.2 4.2.3.21]
+    # HIP allows this intrinsically so this will be ignored silently.
+    parser.add_argument("-extended-lambda", "--extended-lambda",
+                        action='store_true')
+
+    # nvcc allows -std=c++ mixed with C files which in case the option is
+    # ineffective for C files.
+    parser.add_argument(
+        '-std', '--std', choices=['c++03', 'c++11', 'c++14', 'c++17', 'c++12'],
+        help="Specify language standard for CUDA and C++ inputs.")
+
+    # The nvcc differs from clang, hipcc and other compilers by that the
+    # option applies globally and only the last option is considered.
+    parser.add_argument('-x', '--x', choices=['c', 'c++', 'cu'],
+                        help="Specify language of the input files")
+
+    parser.add_argument("-default-stream", "--default-stream",
+                        choices=['legacy', 'null', 'per-thread'])
+
+    parser.add_argument("--version", action='store_true')
+    parser.add_argument("-v", "--verbose", action='store_true')
+
+    # TODO: define __CUDA_ARCH__ and __CUDA_ARCH_LIST__ based on
+    #       '-arch' values.
+    parser.add_argument("-arch", action=IgnoredOption)
+    # TODO: handle -fmad. Investigate what is the nvcc's default for this too.
+    parser.add_argument('-fmad', '--fmad', choices=['true', 'false'],
+                        action=IgnoredOption)
+    parser.add_argument("-maxrregcount", "--maxrregcount",
+                        action=IgnoredOption, help="Ignored option.")
+    parser.add_argument("--keep", nargs=0, action=IgnoredOption)
+    parser.add_argument("--keep-dir", action=IgnoredOption)
+    parser.add_argument("--generate-code", action=IgnoredOption)
+    parser.add_argument("-G", "--device-debug", nargs=0, action=IgnoredOption,
+                        help="Generate debug information for device code.")
+    # "The compiler has an option (-use_fast_math) that forces each
+    # function in Table 9 to compile to its intrinsic counterpart. "
+    # [CUDA PM v12.2 13.2].
+    #
+    # CUDA intrinsics have defined numerical accuracy. This means -ffast-math
+    # is potentially more relaxed than --use_fast_math so we may not use it.
+    parser.add_argument("-use_fast_math", "--use_fast_math",
+                        nargs=0, action=IgnoredOption)
+
+    return parser
+
+def get_hip_path() -> str:
+    """Get HIP path
+    """
+    return "@CUCC_HIP_PATH@"
```

Comment:
Seems like this is the only place where cmake `@` substitution is used. Seems unnecessary as we can pull this information from `.hipInfo`. Furthermore, it's a bit confusing that this file is named .in whereas it's 99% a python file. 

Also, why not allow for switching HIP install via HIP_PATH environment variable?

Reply:
It's a start, not a final form.

> Furthermore, it's a bit confusing that this file is named .in whereas it's 99% a python file.

On the contrary, to me it seems the naming had the preferable outcome: you noticed the file wasn't complete python file.

Reply:
If we use `.hipInfo` then we don't need substitution - I think it makes more sense do it it that way since other tools such as hipcc and hipconfig also rely on hipinfo

Reply:
It's your call.